storage/storage.go

```diff
@@ -289,6 +293,49 @@ func DeleteMod(ctx context.Context, modID string, name string, versionID string)
 	return true
 }
 
+func DeleteMod(ctx context.Context, modID string, name string, versionID string) bool {
+	if storage == nil {
+		return false
+	}
+
+	cleanName := cleanModName(name)
+
+	query := postgres.GetModVersion(ctx, modID, versionID)
+
+	if len(query.Arch) != 0 {
+		for _, link := range query.Arch {
+			if success := DeleteModLink(ctx, modID, cleanName, versionID, link.Platform); !success {
+				return false
+			}
+		}
+	} else {
+		key := fmt.Sprintf("/mods/%s/%s.smod", modID, cleanName+"-"+versionID)
+
+		if err := storage.Delete(key); err != nil {
+			log.Ctx(ctx).Err(err).Msg("failed to delete version")
+			return false
+		}
+	}
+
+	return true
+}
+
+func DeleteModLink(ctx context.Context, modID string, name string, versionID string, platform string) bool {
```

Comment:
Should be renamed to `DeleteModArch`

Reply:
https://github.com/satisfactorymodding/smr-api/pull/12/commits/bdd565d8997568955b8e33bcb352c38c7af5c2c5